EMS-UI/src/app/validation-report/validation-report.component.ts

```diff
@@ -67,10 +70,28 @@ export class ValidationReportComponent implements OnInit {
     this.auditService.getServiceDefinitionSearchAudits()
       .then(
         interactions => {
+          interactions
+            .forEach(int => int.interactionType = InteractionType.SERVICE_SEARCH);
           this.interactions = this.interactions.concat(interactions);
           this.loadedSearchAudits = true;
         }
         //TODO: handle errors properly
       ).catch(err => this.loadedSearchAudits = true);
   }
+
+  private groupByCase(interactions: Interaction[]) : Interaction[] {
+    let grouped = interactions.reduce(function(r, a) {
+      r[a.additionalProperties["caseId"]] = r[a.additionalProperties["caseId"]] || [];
+      r[a.additionalProperties["caseId"]].push(a);
+      return r;
+    }, Object.create(null));
```

Comment:
The intent behind this is probably better captured by a Map object, which seems to be the replacement for this idiom

Reply:
Updated this in latest commit, think it's much cleaner now.

Reply:
Agreed, I like the single-pass approach :)